deepchecks/nlp/checks/multivariate_embeddings_drift_utils.py

```diff
@@ -0,0 +1,120 @@
+from deepchecks.utils.plot import DEFAULT_DATASET_NAMES
+
+SAMPLES_FOR_REDUCTION_FIT = 1000
+
+
+def run_multivariable_drift_for_embeddings(train_dataset: TextData, test_dataset: TextData,
+                                           sample_size: int, random_state: int, test_size: float,
+                                           min_meaningful_drift_score: float, num_samples_in_display: int,
+                                           dimension_reduction_method: str, with_display: bool,
+                                           dataset_names: Tuple[str] = DEFAULT_DATASET_NAMES):
+    """Calculate multivariable drift on embeddings."""
+    # sample train and test datasets equally
+    train_sample = train_dataset.sample(sample_size, random_state=random_state)
+    test_sample = test_dataset.sample(sample_size, random_state=random_state)
+
+    train_sample_df = train_sample.embeddings
+    test_sample_df = test_sample.embeddings
+
+    # create new dataset, with label denoting whether sample belongs to test dataset
+    domain_class_df = pd.concat([train_sample_df, test_sample_df]).reset_index(drop=True)
+    domain_class_labels = pd.Series([0] * len(train_sample_df) + [1] * len(test_sample_df))
+
+    # reduce dimensionality of embeddings if needed.
+    # skips if not required ('none') or if number of features is small enough (< 30) in 'auto' mode.
+    if not (dimension_reduction_method == 'none' or (
+            dimension_reduction_method == 'auto' and domain_class_df.shape[1] < 30)):
+
+        if (dimension_reduction_method == 'auto' and with_display) or dimension_reduction_method == 'umap':
+            reducer = UMAP(n_components=10, n_neighbors=5, init='random', random_state=random_state)
+        else:  # Faster, but graph will look bad.
+            reducer = PCA(n_components=10, random_state=random_state)
+
+        samples_for_reducer = min(SAMPLES_FOR_REDUCTION_FIT, len(domain_class_df))
+        reducer.fit(domain_class_df.sample(samples_for_reducer, random_state=random_state))
+        domain_class_df = pd.DataFrame(reducer.transform(domain_class_df), index=domain_class_df.index)
+
+        # update train and test samples with new reduced embeddings (used later in display)
+        new_embeddings_train = domain_class_df.iloc[:len(train_sample_df)]
+        new_embeddings_test = domain_class_df.iloc[len(train_sample_df):]
+        train_sample.set_embeddings(new_embeddings_train, verbose=False)
+        test_sample.set_embeddings(new_embeddings_test, verbose=False)
+
+    x_train, x_test, y_train, y_test = train_test_split(floatify_dataframe(domain_class_df), domain_class_labels,
+                                                        stratify=domain_class_labels, random_state=random_state,
+                                                        test_size=test_size)
+
+    # train a model to disguise between train and test samples
+    domain_classifier = GradientBoostingClassifier(max_depth=2, random_state=random_state)
```

Comment:
How did you determine depth 2? That's what we use in the tabular version?

Reply:
Yes. I think we lowered it to save time on calculations... Though not sure we need it. Will check how it affects.

Reply:
Doesn't seem to make a significant change. makes sense - there are only 10 features left after reduction, and this is Gradient Boosting which is less sensitive to depth of tree. I think we can pin this as a future research but I don't think it should be change ATM